feat: enhance /addskill talkAction, add new /setskill talkaction

cleberma opened the pull request:
# Description

This PR changes how /addskill works now being more intuitive and adding a new feature called /setskill where an exact skill value can be set

## Behaviour
### **Actual**

The /addskill works poorly and doesn't have a exact skill set function.

### **Expected**

The addskill works in a very intuitive way, calculates everything with correct math and adds a new useful feature.

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested

Testing can be easily done by login a test character and executing parameters.

  - [x] Test A /addskill playerName, skill, 99 (expected behaviour: actual skillLevel + 99 = final skillLevel)

  - [x] Test B /setskill playerName, magic, 99
(expected behaviour: skillLevel goes to the desired flat value = 99)

**Test Configuration**:

  - Server Version: Canary 3.2.0
  - Client: CipSoft 14.05
  - Operating System: Windows 10

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] My changes generate no new warnings
  - [x] I have added tests that prove my fix is effective or that my feature works

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced `/setskill` command to set exact skill levels, magic levels, or player levels.
  * Expanded skill name aliases: use "distance," "shielding," and "fishing" instead of abbreviated forms.

* **Improvements**
  * Enhanced `/addskill` command with support for level advancement, magic level progression, and regular skill increments.
  * Improved parameter validation and error messaging for skill commands.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

cleberma committed:
Enhance /addskill talkAction and new /setskill talkaction

# Description

This PR changes how /addskill works now being more intuitive and adding a new feature called /setskill where an exact skill value can be set

## Behaviour
### **Actual**

The /addskill works poorly, with imprecise and doesn't have a exact skill set function.

### **Expected**

The addskill works in a very intuitive way, easy to catch up, calculates everything with correct math and adds a new very useful feature

## Type of change

  - [x] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested

Testing can be easily done by login a test character and executing parameters.

  - [x] Test A
/addskill comands and /setskill commands like: /addskill playerName, magic, 99
(expected behaviour: actual skillLevel + 99 = final skillLevel)

  - [x] Test B
/setskill playerName, magic, 99
(expected behaviour: actual skillLevel gets shot to 0 and recalculated to the desired flat value: 0 + 99 = 99)

**Test Configuration**:

  - Server Version: Canary 3.2.0
  - Client: CipSoft 14.05
  - Operating System: Windows 10

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [ ] I checked the PR checks reports
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] My changes generate no new warnings
  - [x] I have added tests that prove my fix is effective or that my feature works

gabrielew commented:
`/setskill` -> the `level` option is a bit **dangerous**, as it doesn't take into account the statuses that the player should gain/lose

vllsystems commented:
Perhaps, instead of replacing addskill, include it as setskill.lua and correct some of the messages like Line 71: player:sendCancelMessage("Usage: /addskill <playername>, <skill or 'level'/'magic'>, <amount>")

cleberma commented:
> `/setskill` -> the `level` option is a bit **dangerous**, as it doesn't take into account the statuses that the player should gain/lose

Great observation, i forgot to test the setlevel option as I was more focused on skills.

cleberma committed:
/setskill level, correction.

corrected the well pointed level leak where the level and XP would be changed correctly but health, mana and capacity stats stay the same.
tested every scenario and all seems to be working fine, please review.

Also I don't think the code should be separated in two files because it's the same thing with two different approaches, there's no need to split and duplicate code/files... and performance wouldn't be an issue since the script is only triggered on god talkaction, it's not a big deal.

cleberma committed:
Cleaning

Remove duplicate entries on SkillMap.

cleberma committed:
Merge branch 'opentibiabr:main' into addskill-Talkaction-Enhancement

github-actions[bot] committed:
Lua code format - (Stylua)

dudantas commented:
Hey @cleberma, do you think you could fix the issue cited here? Please give-me a feedback, or I will close the PR.

cleberma commented:
> Hey @cleberma, do you think you could fix the issue cited here? Please give-me a feedback, or I will close the PR.

Sorry, @dudantas but I stopped using PCs because I was addicted to games and content that was bad for me, so I gave up "tech" trying to reconnect to myself.
